app/models/fund.rb

```diff
@@ -0,0 +1,60 @@
+class Fund
+  attr_reader :code
+
+  class InvalidActivity < StandardError; end
+  class InvalidFund < StandardError; end
+
+  def initialize(id)
+    @code = self.class.codelist.find { |c| c["code"] == id }
+
+    raise InvalidFund if @code.nil?
+  end
+
+  def id
+    code["code"]
+  end
+
+  def name
+    code["name"]
+  end
+
+  def gcrf?
+    id == 2
+  end
+
+  def newton?
+    id == 1
+  end
+
+  class << self
+    def from_activity(activity)
+      raise InvalidActivity unless activity.fund?
+      raise InvalidFund unless valid_fragments.include?(activity.roda_identifier_fragment)
+
+      case activity.roda_identifier_fragment
+      when "NF"
+        new(1)
```

Comment:
I spent some time being confused at why we'd use "magic numbers" when we have them in a codelist. It's much easier to make a mistake when setting or comparing with a number than when using their meaningful names or identifiers.

Perhaps this is part of "future optimisations", in which case ignore me! But if identifiers continue to play a big part in the whole thing, could we store the valid identifiers in the codelist, too?

Reply:
Yes, it would at least be helpful to pull these into named constants on the `Fund` model for readability, and perhaps map that back into the migration in the previous commit?

Reply:
Done and done! 👍